contrib/Shopify/sarama/option.go

```diff
@@ -17,3 +19,19 @@ func WithServiceName(name string) Option {
 		cfg.serviceName = name
 	}
 }
+
+// WithAnalytics enables Trace Analytics for all started spans.
+func WithAnalytics(on bool) Option {
+	if on {
+		return WithAnalyticsRate(1.0)
+	}
+	return WithAnalyticsRate(0.0)
+}
+
+// WithAnalyticsRate sets the sampling rate for Trace Analytics events
+// correlated to started spans.
+func WithAnalyticsRate(rate float64) Option {
```

Comment:
It could, but it doesn't have to be. We don't really check it anywhere else either. If the rate is bigger than 1 it would always be kept (see [here](https://github.com/DataDog/datadog-agent/blob/84b29ad6aa0b127c25c429792b2eaeb7f63a1df0/pkg/trace/sampler/score.go#L18-L21)), if it's lower than 0 it will always be rejected (see [same place](https://github.com/DataDog/datadog-agent/blob/84b29ad6aa0b127c25c429792b2eaeb7f63a1df0/pkg/trace/sampler/score.go#L18-L21))

Reply:
OK what is important is that the Agent has that.